### ISSUE TITLE
Builder simulator for "native" verification

### ISSUE THREAD

**Issue by ledwards2225** [AztecProtocol/barretenberg#673]
For Honk, we'd like to have a single Verifier class for both recursive and "native" verification. Since the recursive verifier is the first class object, the logic will be set up to work for stdlib types and we will achieve native verification via swapping a genuine builder (e.g. `UltraCircuitBuilder`) for a "simulated" builder, a la Cody's work. Once this is done, verifier classes like SumcheckVerifier, GeminiVerifier, ShplonkVerifier and UltraVerifier will only every be instantiated with stdlib types. This will eliminate the need for the constexpr branching (e.g. on `IsRecursiveFlavor` or `Curve::is_stdlib_type`) in these classes which currently facilitates supporting both stdlib and native logic. Instead, these classes will be templated over the appropriate recursive concept, e.g. `RecursiveFlavor`.

**Comment by codygunton** [comment]
The CircuitSimulator was scrapped as being lower priority now that the rollup logic will be written in Noir. Closing this, at least for now.
